MapboxNavigation/BottomBannerView.swift

```diff
@@ -0,0 +1,136 @@
+import UIKit
+import MapboxCoreNavigation
+import MapboxDirections
+
+protocol BottomBannerViewDelegate: class {
+    func didCancel()
+}
+
+@IBDesignable
+@objc(MBBottomBannerView)
+open class BottomBannerView: UIView {
+    
+    weak var timeRemainingLabel: TimeRemainingLabel!
+    weak var distanceRemainingLabel: DistanceRemainingLabel!
+    weak var arrivalTimeLabel: ArrivalTimeLabel!
+    weak var cancelButton: CancelButton!
+    weak var separatorView: SeparatorView!
+    
+    let dateFormatter = DateFormatter()
+    let dateComponentsFormatter = DateComponentsFormatter()
+    let distanceFormatter = DistanceFormatter(approximate: true)
+    let routeStepFormatter = RouteStepFormatter()
+    weak var routeController: RouteController!
+    
+    weak var delegate: BottomBannerViewDelegate?
+    
+    var congestionLevel: CongestionLevel = .unknown {
+        didSet {
+            switch congestionLevel {
+            case .unknown:
+                timeRemainingLabel.textColor = timeRemainingLabel.trafficUnknownColor
+            case .low:
+                timeRemainingLabel.textColor = timeRemainingLabel.trafficLowColor
+            case .moderate:
+                timeRemainingLabel.textColor = timeRemainingLabel.trafficModerateColor
+            case .heavy:
+                timeRemainingLabel.textColor = timeRemainingLabel.trafficHeavyColor
+            case .severe:
+                timeRemainingLabel.textColor = timeRemainingLabel.trafficSevereColor
+            }
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        distanceFormatter.numberFormatter.locale = .nationalizedCurrent
+        dateFormatter.timeStyle = .short
+        dateComponentsFormatter.allowedUnits = [.hour, .minute]
+        dateComponentsFormatter.unitsStyle = .abbreviated
+        
+        setupViews()
+        setupLayout()
+        
+        cancelButton.addTarget(self, action: #selector(BottomBannerView.cancel(_:)), for: .touchUpInside)
+    }
+    
+    @IBAction func cancel(_ sender: Any) {
+        delegate?.didCancel()
+    }
+    
+    override open func prepareForInterfaceBuilder() {
+        super.prepareForInterfaceBuilder()
+        separatorView.backgroundColor = .red
+        timeRemainingLabel.text = "22 min"
+        distanceRemainingLabel.text = "4 mi"
+        arrivalTimeLabel.text = "10:09"
```

Comment:
I assume these are placeholders?

Reply:
Yes, `prepareForInterfaceBuilder()` is only called when rendering through IB.